src/DotNet/Emit/DynamicMethodBodyReader.cs

```diff
@@ -207,6 +207,7 @@ public DynamicMethodBodyReader(ModuleDef module, object obj, Importer importer)
 			reader = ByteArrayDataReaderFactory.CreateReader(code);
 			method = CreateMethodDef(delMethod);
 			parameters = method.Parameters;
+			context = module.Context;
```

Comment:
It should be able to pass this to the base ctor.

Reply:
I'm just following the existing code here. I'm not sure why several of these fields (`reader`, `parameters`, etc) are being set in this way rather than passed to the base constructor.

Reply:
It's because it has to extract various things through reflection and it can't do that in a base() call. It would've been better to turn it into some Create() method instead which does all of that and then calls some internal ctor but it's too late now.

The context can easily be passed to the base, and that will also allow us to make the field readonly.